narwhals/_arrow/expr_struct.py

```diff
@@ -0,0 +1,24 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from narwhals._expression_parsing import reuse_series_namespace_implementation
+
+if TYPE_CHECKING:
+    from typing_extensions import Self
+
+    from narwhals._arrow.expr import ArrowExpr
+
+
+class ArrowExprStructNamespace:
+    def __init__(self: Self, expr: ArrowExpr) -> None:
+        self._compliant_expr = expr
+
+    def field(self: Self, name: str) -> ArrowExpr:
+        return reuse_series_namespace_implementation(
+            self._compliant_expr,
+            "struct",
+            "field",
+            name=name,
+            evaluate_output_names=lambda _col: [name],
```

Comment:
For all implementations, instead of overwriting `evaluate output names`,  can we just use `.alias`?

A good test would be `nw.col('a').struct.field('b').name.keep()`. I think for polars the resulting column name would be 'a' we should check that we do the same